docker/credentials_helper_test.go

```diff
@@ -2,26 +2,31 @@ package docker
 
 import "testing"
 
-func Test_to2ndLevelDomain(t *testing.T) {
+func Test_registryEquals(t *testing.T) {
 	tests := []struct {
-		name   string
-		rawurl string
-		want   string
+		name string
+		urlA string
+		urlB string
+		want bool
 	}{
-		{"2nd level", "quay.io", "quay.io"},
-		{"3nd level", "sub.quay.io", "quay.io"},
-		{"localhost", "localhost", "localhost"},
-		{"2nd level with protocol", "https://docker.io", "docker.io"},
-		{"2nd level with path", "docker.io/v1/", "docker.io"},
-		{"2nd level with port", "docker.io:80", "docker.io"},
-		{"2nd level with protocol and path", "https://docker.io/v1/", "docker.io"},
-		{"3rd level with protocol and path", "https://index.docker.io/v1/", "docker.io"},
-		{"3rd level with protocol and path and port", "https://index.docker.io:80/v1/", "docker.io"},
-		{"localhost with protocol and path and port", "http://localhost:8080/v1/", "localhost"},
+		{"no port matching host", "quay.io", "quay.io", true},
+		{"non-matching host different sub-domain", "sub.quay.io", "quay.io", false},
+		{"non-matching host different sub-domain", "sub.quay.io", "sub3.quay.io", false},
+		{"localhost", "localhost", "localhost", true},
+		{"localhost with standard ports", "localhost:80", "localhost:443", true},
+		{"localhost with matching port", "https://localhost:1234", "http://localhost:1234", true},
+		{"localhost with match by default port", "http://localhost", "localhost:80", true},
+		{"localhost with match by default port", "https://localhost", "localhost:443", true},
+		{"localhost with mismatch by default port", "https://localhost", "http://localhost", true},
+		{"docker.io matching host", "https://docker.io", "docker.io", true},
+		{"docker.io matching host", "http://docker.io", "docker.io", true},
```

Comment:
Are these good?

Reply:
As above, a few very minor nitpicks to keep test case descriptions unique.
```suggestion
		{"localhost with standard ports", "localhost:80", "localhost:443", true},
		{"localhost with matching port", "https://localhost:1234", "http://localhost:1234", true},
		{"localhost with match by default port 80", "http://localhost", "localhost:80", true},
		{"localhost with match by default port 443", "https://localhost", "localhost:443", true},
		{"localhost with mismatch by default port", "https://localhost", "http://localhost", true},
		{"docker.io matching host https", "https://docker.io", "docker.io", true},
		{"docker.io matching host http", "http://docker.io", "docker.io", true},
```
As for whether or not 80 == 443, I guess it makes sense if the registry service upgrades/downgrades automatically. But I'm not completely sure of the intent here. Maybe some comments in the test code would be helpful.